### PR TITLE
Update Developer Guide

### PR DESCRIPTION
**Pull request by DonnyS57**
Updated the documentation for Model.

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -125,6 +125,7 @@ The `Model` component,
 * stores the currently 'selected' `Person` objects (e.g., results of a search query) as a separate _filtered_ list which is exposed to outsiders as an unmodifiable `ObservableList<Person>` that can be 'observed' e.g. the UI can be bound to this list so that the UI automatically updates when the data in the list change.
 * stores a `UserPref` object that represents the user’s preferences. This is exposed to the outside as a `ReadOnlyUserPref` objects.
 * does not depend on any of the other three components (as the `Model` represents data entities of the domain, they should make sense on their own without depending on other components)
```
Comment (bharathcs) [inline docs/DeveloperGuide.md:127]: This part isn't quite right anymore, I believe

**Review by bharathcs — CHANGES_REQUESTED** [review]
Might need to look at the list of bullet points, now that Model Layer takes in storage and needs it to save.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (DonnyS57) [reply docs/DeveloperGuide.md]: I fixed it!

**Review by bharathcs — APPROVED** [review]
LGTM!

**Pull request merged**